client/src/app/site/agenda/components/agenda-sort/agenda-sort.component.ts

```diff
@@ -155,4 +156,15 @@ export class AgendaSortComponent extends BaseSortTreeComponent<ViewAgendaItem> i
                 return 'visibility_off';
         }
     }
+
+    private getNumericValueForAgendaItemVisibility(visibility: AgendaItemVisibility): number {
```

Comment:
THis seems hacks. Can you adopt the filter to not use numeric values but strings?

Reply:
This would be some more effort. Therefore, I create a new issue for that.